libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -246,7 +247,7 @@ internal class MapboxTripSession(
 
         this.rawLocation = rawLocation
         locationObservers.forEach { it.onNewRawLocation(rawLocation) }
-        mainJobController.scope.launch {
+        mainJobController.scope.launch(Dispatchers.Unconfined) {
```

Comment:
This is a cool idea 👍 

However, I think that we do want the coroutine to be confined to the main thread in case Nav Native would start returning the `updateLocation` on a worker thread (for some reason). The rest of our code assumes here that we run on the main thread, so we need to remain confined.

Reading docs further, looks like we can do
```suggestion
        mainJobController.scope.launch(start = CoroutineStart.UNDISPATCHED) {
```
to achieve the same effect of running the coroutine until suspension in the current call frame which is the goal, while keeping the result confined to the main thread.

Reply:
Thank you for suggestion. Applied 👍